sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
-            matplotlib.pylab.figure()
-        except:
+            import matplotlib.pyplot as plt
+            plt.figure()
+        except ImportError:
```

Comment:
Sorry if I was unclear. I understand you point and I also don't see why the two lines are necessary. Maybe @pprett can enlighten us?